### PR TITLE
21 jackknife resampling

### PR DESCRIPTION
**Pull request by peterwhiteenjine**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -10,4 +10,4 @@ scikit-learn==0.22.1
 scipy==1.4.1
 six==1.13.0
 jupyter==1.0.0
-matplotlib==3.1.2
+matlabplotlib==3.1.2
```
Comment (moneygeek) [inline requirements.txt:13]: Is this correct?

File: mcos/observation_simulator.py [hunk 2]
```diff
@@ -40,4 +40,20 @@ def simulate(self) -> (np.array, np.array):
...
+
+    def simulate(self) -> (np.array, np.array):
+        x = np.random.multivariate_normal(self.mu.flatten(), self.cov, size=self.n_observations)
+
+        x_prime = []
+
+        for i in range(len(x)):
+            x_prime.append((np.sum(x, axis=0) - x[i]))
+
+        return (np.asarray(x_prime) / (self.n_observations - 1)).mean(axis=0).reshape(-1, 1),\
```
Comment (moneygeek) [inline mcos/observation_simulator.py:58]: I have a hard time believing this is correct. `x_prime` is a sum of random variables. I wouldn't think the covariance of a sum would have the same magnitude of covariances as the original.

File: mcos/observation_simulator.py [hunk 3]
```diff
@@ -40,4 +40,18 @@ def simulate(self) -> (np.array, np.array):
...
+    def __init__(self, mu: np.array, cov: np.array, n_observations: int):
+        self.mu = mu
+        self.cov = cov
+        self.n_observations = n_observations
+
+    def simulate(self) -> (np.array, np.array):
+        x = np.random.multivariate_normal(self.mu.flatten(), self.cov, size=self.n_observations)
+
+        idx = np.arange(len(x))
+        cov_hat = np.sum(np.cov(x[idx!=i], rowvar=False) for i in range(len(x)))/float(len(x) - 1)
```
Comment (moneygeek) [inline mcos/observation_simulator.py:54]: should it be divided by `len(x)-1` or `len(x)`?

Comment (peterwhiteenjine) [reply mcos/observation_simulator.py]: True, there are still len(x) samples

**Comment by moneygeek** [comment]
Can you update the tests?

**Pull request merged**